ctpolicy/ctpolicy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// GroupBuilder owns information on log group membership logic.
+type GroupBuilder interface {
+	GetName() string
+	GetDescription() string
+	IsMember(log *loglist.Log) bool
+	IsBase() bool // whether group covers all logs
+}
+
+// LogGroup holds information on a single log group.
+type LogGroup struct {
+	name        string
+	description string
+	logURLMap   map[string]bool
+	isBase      bool
+}
+
+// BuildGroup makes an instance of LogGroup using builder and Logs provided.
+func BuildGroup(gb GroupBuilder, ll *loglist.LogList) LogGroup {
+	var lg LogGroup
+	lg.name = gb.GetName()
+	lg.description = gb.GetDescription()
+	lg.logURLMap = make(map[string]bool)
+	for _, l := range ll.Logs {
+		if gb.IsMember(&l) {
+			lg.logURLMap[l.URL] = true
+		}
+	}
+	lg.isBase = gb.IsBase()
+	return lg
+}
+
+// CTPolicy describes requirements determined for logs in terms of per-group submits.
+type CTPolicy interface {
+	// How many logs required per group to satisfy the policy.
+	GroupMinInclusions(cert *x509.Certificate) map[string]int
+	// How many logs could be requested per group implying concurrent requests.
+	// If not specified, falls back intoGroupMinInclusions.
```

Comment:
Do the CT policies have anything to say about numbers of concurrent submissions?  (If not, this feels like it's mixing up what the policy requires with operational heuristics for how to achieve those requirements.)

Reply:
Dropped concurrency issue out of this PR.